### ISSUE TITLE
Sort by relevance is not supported for full text search.

### ISSUE THREAD

**Issue by lokesh-everest** [feathersjs-ecosystem/feathers-mongoose#331]
I have few fields on a collection with full-text search index on. But I cant search and get the most relevant results by sorting them with text score. 

I am trying to add this to query param:
{$sort" :{ score: { $meta: "textScore" } }

It's giving this me this error 
"must have $meta projection for all $meta sort keys"

Is there any way I can get around this?

**Solution**:
If I change In service.js in `feathers-mongoose` const q = model.find(query).lean(this.lean);
to 

```
const q = model
			.find(query, {
				score: { $meta: "textScore" }
			})
			.lean(this.lean)
```

It's working. There should be an option to send my options to find like the way they node MongoDB driver allows. 
http://mongodb.github.io/node-mongodb-native/3.0/api/Collection.html#find

As we are in the production application we need this badly. Please respond as soon as possible.

**Comment by jamesjnadeau** [comment]
Hi, I'm not a lead dev here, but I've helped out on occasion.

I believe that this is "driver" specific, and not like the other DB drivers out there, so I don't imagine this will be fixed or accepted, but again, i'm not in charge here. Probably would make a good configuration option....

The beauty of feathers is this is easy for you to override/'monkey patching this to work how you want it to.

The code would look something like(not tested/complete):
```
var feathersMongoose = require('feathers-mongoose');
var service = feathersMongoose({
  Model: model,
});
service._find = function(params = {}) {
// put the correct functionality you need here
// see https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/master/lib/service.js#L48
}
```

I've done this a lot with SQL services where I had to customize the query used. It's pretty common and easy to maintain over time in my experience.
